src/client/src/containers/SelectOption/index.tsx

```diff
@@ -144,6 +144,27 @@ class SelectOption extends React.Component<Props, ISelectOptionState> {
     });
   }
 
+  public removeOption(
+    cardNumber: number,
+    cardCount: number,
+    internalName: string
+  ) {
+    const { selectedCardIndices, currentCardData, selectOptions } = this.props;
+    if (selectOptions && currentCardData && currentCardData.length > 1) {
+      const size = currentCardData.length;
+      for (let i = size - 1; i >= 0; i--) {
+        if (currentCardData[i].internalName === internalName) {
+          currentCardData.splice(i, 1);
```

Comment:
I think `splice` might not be good here as the `currentCardData` is part of the state (`this.props`) and should be immutable/constant.

Reply:
If it is manipulating the state object, it is definitely needs to be corrected. However, please guide me if I am wrong. In select pages, I guess, we already destructure the state object before returning them as props. So, I think we do not need to create a copy of element here. 
Am I thinking right?

Reply:
This eventually creates a new object state, but it is still best practice not to touch the current state. It should be a copy of the array.